presto-hive/src/main/java/io/prestosql/plugin/hive/HiveTableHandle.java

```diff
@@ -206,13 +206,31 @@ public boolean equals(Object o)
         }
         HiveTableHandle that = (HiveTableHandle) o;
         return Objects.equals(schemaName, that.schemaName) &&
-                Objects.equals(tableName, that.tableName);
+                Objects.equals(tableName, that.tableName) &&
+                Objects.equals(tableParameters, that.tableParameters) &&
+                Objects.equals(partitionColumns, that.partitionColumns) &&
+                Objects.equals(partitions, that.partitions) &&
+                Objects.equals(compactEffectivePredicate, that.compactEffectivePredicate) &&
```

Comment:
Some of these fields (e.g., compactEffectivePredicate, enforcedPredicate) are not part of the identity of a hive "virtual" table. Handles should be equal to each other if they represent the same "object" (handles should be thought of like opaque IDs).

Reply:
If a handle represents table+predicate, it's no longer "the same" as handle representing a table.
Later on we will likely introduce table handles modelling a pushed down join.

Maybe we should remove equals from table handles? And treat them truly opaque?

Reply:
@martint @electrum your feedback would be appreciated.

Reply:
> Maybe we should remove equals from table handles? And treat them truly opaque?

They are opaque (from the point of view that the engine doesn't care about what's inside), but just like an id, you should be able to test `id1 == id2` to see if they are the same. At some point we'll need that for some forms of memoization in the optimizer.

Reply:
Okay, so they are opaque but need to implement equals().

- we need document this requirement at SPI interface level
- `HiveTableHandle` as it is today doesn't meet memo's requirements, since it ignores part of its (important) state in equals

if you agree, we should proceed with the PR, right?